### PR TITLE
Bugfix and add Low Power Mode

### PR DESCRIPTION
**Pull request by hallard**
- Created Low Power Battery Monitor example enabling lora module sleep mode
- Added Hard reset of Lora module at startup (with IO pin)
- Fixed issue #9 
- Enable Low Power of Lora Module of other examples sketches

### CODE REVIEW

**Comment by hallard** [comment]
This new pull request #12 should fix your issue, just let me know

File: library.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name=TheThingsNode
-version=2.0.2
+version=2.0.3
```
Comment (johanstokking) [inline library.properties]: We're bumping versions in the `master` branch after merging – sometimes we combine PRs in a version

File: src/TheThingsNode.cpp [hunk 2]
```diff
@@ -72,9 +73,12 @@ void TTN_BUTTON_FN()
   uint8_t trigger = getPinChangeInterruptTrigger(digitalPinToPCINT(TTN_BUTTON));
   if (trigger == FALLING)
   {
+    //Serial.println(F("IPRESS"));
```
Comment (johanstokking) [inline src/TheThingsNode.cpp]: Remove if not used

File: src/TheThingsNode.cpp [hunk 3]
```diff
@@ -217,8 +221,16 @@ void TheThingsNode::loop()
   }
   else
   {
-    Serial.flush();
-    deepSleep();
+    // Don't go to sleep mode while button is still pressed
+    // because if so, timer of ms will be stopped and duration
+    // of pressed button will not work
+    if ( this->buttonPressed ) {
```
Comment (johanstokking) [inline src/TheThingsNode.cpp]: Formatting (spaces)

File: src/TheThingsNode.cpp [hunk 4]
```diff
@@ -737,6 +749,15 @@ TheThingsNode::TheThingsNode()
   pinMode(TTN_BLUE_LED, OUTPUT);
   setColor(TTN_BLACK);
 
+  // set pin mode every time to avoid constraining the user about when the pin is initialized
+  #ifdef TTN_LORA_RESET
```
Comment (johanstokking) [inline src/TheThingsNode.cpp]: Formatting (indentation)

File: docs/TheThingsNode.md [hunk 5]
```diff
@@ -81,6 +81,9 @@ void sleep() {
 }
 ```
 
+Pay attention, this internal sleep method of the library does not put the Lora module (RN2483) into sleep mode and thus your node may consume 3mA even in sleep mode. You need to manually set the lora module to sleep and wake.
```
Comment (johanstokking) [inline docs/TheThingsNode.md]: Lora -> LoRa
lora -> LoRa

File: examples/Basic/Basic.ino [hunk 6]
```diff
@@ -140,4 +143,11 @@ void sendData(uint8_t port)
   payload[5] = bytes[0];
 
   ttn.sendBytes(payload, sizeof(payload), port);
+
+  // Set RN2483 to sleep mode
+  ttn.sleep( 60000 );
```
Comment (johanstokking) [inline examples/Basic/Basic.ino]: Formatting (spaces)

File: examples/BatteryMonitorLPP/BatteryMonitorLPP.ino [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+	  debugSerial.print(F("Button:\t"));
+	  debugSerial.print(value);
+	  debugSerial.println(F("ms"));
+	  lpp.addAnalogInput(7, value/1000.0);
+  }
+
+  ttn.sendBytes(lpp.getBuffer(), lpp.getSize(), port);
+
+  // Set RN2483 to sleep mode
+  ttn.sleep( (uint32_t) (CONFIG_INTERVAL*1000) );
```
Comment (johanstokking) [inline examples/BatteryMonitorLPP/BatteryMonitorLPP.ino]: Formatting (spaces)

File: examples/BatteryMonitorLPP/BatteryMonitorLPP.ino [hunk 8]
```diff
@@ -0,0 +1,172 @@
+// **********************************************************************************
```
Comment (johanstokking) [inline examples/BatteryMonitorLPP/BatteryMonitorLPP.ino]: Although useful, we're not using these headers in examples.

Please add the concerning comments in code, if necessary.

File: docs/TheThingsNode.md (hunk 5 above)
Comment (johanstokking) [reply docs/TheThingsNode.md]: RN2483 can also be RN2903

**Review by johanstokking — CHANGES_REQUESTED** [review]
Thanks @hallard !! Looks good. Few changes, mostly styling. I'd like to keep this repository clean and consistent.

File: examples/BatteryMonitorLPP/BatteryMonitorLPP.ino [hunk 9]
```diff
@@ -122,7 +122,7 @@ void sendData(uint8_t port, uint32_t value)
   ttn.sendBytes(lpp.getBuffer(), lpp.getSize(), port);
 
   // Set RN2483 to sleep mode
-  ttn.sleep( (uint32_t) (CONFIG_INTERVAL*1000) );
+  ttn.sleep( CONFIG_INTERVAL*1000 );
```
Comment (johanstokking) [inline examples/BatteryMonitorLPP/BatteryMonitorLPP.ino]: I meant to say no spaces in the parentheses 😊

**Comment by johanstokking** [comment]
Can you also fix the CI?

**Comment by hallard** [comment]
I've done all modifications, travis should be fine on next check

**Comment by johanstokking** [comment]
@hallard you're using the `wake()` and `sleep()` functions which are not in the referenced release for testing of `TheThingsNetwork`.

Please change .travis.yml#L14 to version `v2.5.6`

**Comment by hallard** [comment]
It's done, sorry wrong comments
Requires TheThingsNode library 2.5.6
should be 
Requires TheThingsNetwork library 2.5.6

**Review by johanstokking — APPROVED** [review]

**Comment by johanstokking** [comment]
Thanks, this is now part of https://github.com/TheThingsNetwork/arduino-node-lib/releases/tag/v2.0.5

**Pull request merged**